middleman_protocol/tests/test_stream_async.py

```diff
@@ -0,0 +1,84 @@
+from asyncio import StreamReader
+from asyncio import StreamWriter
+
+import pytest
+from assertpy import assert_that
+from golem_messages.message import Ping
+from mock import Mock
+
+from middleman_protocol import constants
+from middleman_protocol import exceptions
+from middleman_protocol.constants import ESCAPE_CHARACTER
+from middleman_protocol.constants import ESCAPE_SEQUENCES
+from middleman_protocol.constants import FRAME_SEPARATOR
+from middleman_protocol.message import GolemMessageFrame
+from middleman_protocol.stream import append_frame_separator
+from middleman_protocol.stream import escape_encode_raw_message
+from middleman_protocol.stream_async import handle_frame_receive_async
+from middleman_protocol.stream_async import map_exception_to_error_code
+from middleman_protocol.stream_async import send_over_stream_async
+from middleman_protocol.testing_utils import async_stream_actor_mock
+
+from .utils import generate_ecc_key_pair
+
+
+(CONCENT_PRIVATE_KEY, CONCENT_PUBLIC_KEY) = generate_ecc_key_pair()
+
+
+def _run_test_in_event_loop(event_loop, coroutine, *args):
+    task = event_loop.create_task(coroutine(*args))
+    event_loop.run_until_complete(task)
+    return task
+
+
+class TestHandleFrameReceiveAsync:
+    def test_that_when_frame_with_escaped_sequence_and_separator_is_received_unescaped_frame_is_returned(self, event_loop):
+        expected_data = GolemMessageFrame(Ping(), 777).serialize(CONCENT_PRIVATE_KEY)
```

Comment:
move serialization to line below and remove it from the last assertion

Reply:
Doesn't work for some reason. Bytes are the same, but Frames are different objects, so assertion raise exception.

Reply:
two options:
1) ```python     def test_that_when_frame_with_escaped_sequence_and_separator_is_received_unescaped_frame_is_returned(self, event_loop):
        golem_message_frame = GolemMessageFrame(Ping(), 777)
        data_to_send = escape_encode_raw_message(golem_message_frame.serialize(CONCENT_PRIVATE_KEY)) + ESCAPE_SEQUENCES[ESCAPE_CHARACTER] + FRAME_SEPARATOR
        mocked_reader = self._prepare_mocked_reader(data_to_send)

        task = _run_test_in_event_loop(event_loop, handle_frame_receive_async, mocked_reader, CONCENT_PUBLIC_KEY)

        assert_that(task.done()).is_true()
        mocked_reader.readuntil.mock.assert_called_once_with(FRAME_SEPARATOR)
        assert_that(task.result().payload).is_equal_to(golem_message_frame.payload)
        assert_that(task.result().payload_type).is_equal_to(golem_message_frame.payload_type)
        assert_that(task.result().request_id).is_equal_to(golem_message_frame.request_id)
```
2. Implement `__eq__` method in `GolemMessageFrame` (and other methods inheriting from `AbstrctFrame)`

Reply:
Done